spock-core/src/main/java/org/spockframework/compiler/WhereBlockRewriter.java

```diff
@@ -75,27 +75,28 @@ private void rewrite() {
   private void rewriteWhereStat(ListIterator<Statement> stats) throws InvalidSpecCompileException {
     Statement stat = stats.next();
     BinaryExpression binExpr = AstUtil.getExpression(stat, BinaryExpression.class);
-    if (binExpr == null || binExpr.getClass() != BinaryExpression.class) // don't allow subclasses like DeclarationExpression
-      notAParameterization(stat);
-
-    int type = binExpr.getOperation().getType();
-
-    if (type == Types.LEFT_SHIFT) {
-      Expression leftExpr = binExpr.getLeftExpression();
-      if (leftExpr instanceof VariableExpression)
-        rewriteSimpleParameterization(binExpr, stat);
-      else if (leftExpr instanceof ListExpression)
-        rewriteMultiParameterization(binExpr, stat);
-      else
+    if (binExpr != null && binExpr.getClass() == BinaryExpression.class) // don't allow subclasses like DeclarationExpression
+    {
+      int type = binExpr.getOperation().getType();
+
+      if (type == Types.LEFT_SHIFT) {
+        Expression leftExpr = binExpr.getLeftExpression();
+        if (leftExpr instanceof VariableExpression)
+          rewriteSimpleParameterization(binExpr, stat);
+        else if (leftExpr instanceof ListExpression)
+          rewriteMultiParameterization(binExpr, stat);
+        else
+          notAParameterization(stat);
+      } else if (type == Types.ASSIGN)
+        rewriteDerivedParameterization(binExpr, stat);
+      else if (getOrExpression(binExpr) != null) {
+        stats.previous();
+        rewriteTableLikeParameterization(stats);
+      } else
         notAParameterization(stat);
-    } else if (type == Types.ASSIGN)
-      rewriteDerivedParameterization(binExpr, stat);
-    else if (getOrExpression(binExpr) != null) {
-      stats.previous();
-      rewriteTableLikeParameterization(stats);
-    }
-    else
+    } else if (!AstUtil.isDataTableSeparator(AstUtil.getExpression(stat, Expression.class))) {
```

Comment:
Maybe add a comment here, generally the AST code could use a bit more comments

Reply:
I right away sprinkled comments into the whole method :-)